### PR TITLE
feat: generates operator's ClusterServiceVersion manifests

### LINKED ISSUE #79: Add Cluster Service Version metadata
https://github.com/operator-framework/operator-lifecycle-manager/blob/00eab85df0fd570891754cc743bc3d5831e9dd62/doc/design/building-your-csv.md

### PR DESCRIPTION
**Pull request by bartoszmajsak**
#### Short description of what this resolves:

Generates operator OLM manifests.

#### Changes proposed in this pull request:

- changes `pkg/apis/istio` to `pkg/apis/maistra`
- `operator-sdk` changes:
  - generating CRDs from the code
  - aligned structure with the one `operator-sdk` scaffolds (otherwise CRDs wouldn't be generated - there's currently no way to change output dir)
  - generates ClusterServiceVersion manifest as part of `operator-codegen` make target
  - we are using Openshit Template for operator deployment, but `operator-sdk` does not understand this `Kind`. I generate `operator.yaml` on the fly whenever `operator.tpl.yaml` is changed
- minor fixes in templates
- bumped `operator-sdk` to `0.17.2`
- renamed cr yamls to comply with csv gen
- minor improvements in `Makefile`

`cluster_role.yaml` and `role.yaml` are identical. This was causing the former to not appear in generated CSV manifest. Shuffling the order apparently "solves" the problem.

Fixes #79

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -363,8 +373,8 @@ deploy-test-%:
 	oc new-project $(TEST_NAMESPACE) || true
 	oc adm policy add-scc-to-user anyuid -z default -n $(TEST_NAMESPACE)
 	oc adm policy add-scc-to-user privileged -z default -n $(TEST_NAMESPACE)
-	oc apply -n $(TEST_NAMESPACE) -f deploy/bookinfo/session_role.yaml
-	oc apply -n $(TEST_NAMESPACE) -f deploy/bookinfo/session_rolebinding.yaml
+	oc apply -n $(TEST_NAMESPACE) -f deploy/examples/bookinfo/session_role.yaml
+	oc apply -n $(TEST_NAMESPACE) -f deploy/examples/bookinfo/session_rolebinding.yaml
```
Comment (aslakknutsen) [inline Makefile:377]: Maybe remove "bookinfo" level or rename it while we're at it?

File: deploy/olm-catalog/istio-workspace/manifests/istio-workspace.clusterserviceversion.yaml [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+              "value": "jason"
+            }
+          }
+        }
+      ]
+    capabilities: Basic Install
+  name: istio-workspace.v0.0.3
+  namespace: placeholder
+spec:
+  apiservicedefinitions: {}
```
Comment (aslakknutsen) [inline deploy/olm-catalog/istio-workspace/manifests/istio-workspace.clusterserviceversion.yaml:31]: WARN[0000] Required csv fields not filled in file istio-workspace.clusterserviceversion.yaml:
	spec.description
	spec.provider

File: Makefile (hunk 1 above)
Comment (bartoszmajsak) [reply Makefile:377]: Yeah, that `bookinfo` thing is irrelevant at this point.

File: deploy/olm-catalog/istio-workspace/manifests/istio-workspace.clusterserviceversion.yaml (hunk 2 above)
Comment (bartoszmajsak) [reply deploy/olm-catalog/istio-workspace/manifests/istio-workspace.clusterserviceversion.yaml:31]: Good point - it was wiped out in subsequent generation I believe :)

File: Makefile (hunk 1 above)
Comment (bartoszmajsak) [reply Makefile:377]: db1b00f

File: deploy/olm-catalog/istio-workspace/manifests/istio-workspace.clusterserviceversion.yaml (hunk 2 above)
Comment (bartoszmajsak) [reply deploy/olm-catalog/istio-workspace/manifests/istio-workspace.clusterserviceversion.yaml:31]: c19d0d8

**Pull request merged**